File: integration-tests/pkg/cluster/cluster.go
package cluster

type Cluster struct {
	Name              string     `yaml:"name"`
	PrivateNetwork    string     `yaml:"privateNetwork"`
	PrivateSubnetwork string     `yaml:"privateSubnetwork"`
	KubernetesVersion string     `yaml:"kubernetesVersion"`
	SkipInitApply     bool       `yaml:"skipInitApply"`
	Machines          []*Machine `yaml:"machines"`
}

type Machine struct {
	ID                     string         `yaml:"id"`
	Type                   string         `yaml:"type"`
	Platform               string         `yaml:"platform"`
	TalosInitialVersion    string         `yaml:"talosInitialVersion"`
	TalosImage             string         `yaml:"talosImage"`
	PrivateIP              string         `yaml:"privateIP"`
	ConfigPatches          []string       `yaml:"configPatches"`
	ApplyConfigViaUserdata bool           `yaml:"apply-config-via-userdata"`
	Hcloud                 *HcloudMachine `yaml:"hcloud"`
}

type HcloudMachine struct {
	ServerType string `yaml:"serverType"`
	Datacenter string `yaml:"datacenter"`
}

File: integration-tests/pkg/cluster/typescript/cluster.ts
import type * as pulumi from "@pulumi/pulumi";

export type HcloudMachine = {
  serverType: string;
  datacenter?: string;
};

export type Machine = {
  id: string;
  type: string;
  platform?: string;
  variant: "cloud" | "metal";
  talosInitialVersion?: string;
  talosImage?: string;
  privateIP: string;
  configPatches?: string[];
  userdata?: string;
  applyConfigViaUserdata?: boolean;
  hcloud?: HcloudMachine;
};

export type Cluster = {
  name: string;
  privateNetwork?: string;
  privateSubnetwork?: string;
  kubernetesVersion: string;
  skipInitApply?: boolean;
  machines: Machine[];
};

export type Server = {
  id(): string;
  ip(): pulumi.Output<string>;
  withUserdata(userdata: pulumi.Output<string>): Server;
};

export type Network = {
  id(): pulumi.Output<string>;
};

export type Deployed = {
  servers: Server[];
  deps: pulumi.Resource[];
};

export type DeployedCluster = {
  clusterMachines: pulumi.Output<Record<string, string>>;
  credentials: DeployedCredentials;
};

export type DeployedCredentials = {
  kubeconfig: pulumi.Output<string>;
  talosconfig: pulumi.Output<string>;
};
